src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc

```diff
@@ -110,14 +111,12 @@ void AddSubMenuRequest::Run() {
                                  [strings::value].asString(),
                       app,
                       application_manager_);
-
     msg_params[strings::sub_menu_icon] =
         (*message_)[strings::msg_params][strings::sub_menu_icon];
-    SendHMIRequest(hmi_apis::FunctionID::UI_AddSubMenu, &msg_params, true);
   } else {
-    LOG4CXX_ERROR(logger_, "Sub-menu icon is not valid.");
-    SendResponse(false, mobile_apis::Result::INVALID_DATA);
+     msg_params.erase(strings::sub_menu_icon);
```

Comment:
@pold500 I think that you should check separately if icon type is STATIC or DYNAMIC, because only in case of DYNAMIC icon you should add full image path.